### PR TITLE
feat: redirects

### LINKED ISSUE #5: Support redirects

### PR DESCRIPTION
**Pull request by dsherret**
Adds support for some redirects:

- `command > file.txt`
- `command > /dev/null` - Supress stdout
- `command 1> file.txt` - Same as default
- `command 2> file.txt` - Redirect stderr
- `command &> file.txt` - Redirect both stdout and stderr

Does not implement `&>` redirects or multiple redirects.

Closes #5

### CODE REVIEW

**Comment by dsherret** [comment]
@kt3k I'm going to have to do a much larger refactor so we can get started on this properly. I'll do it tomorrow.

**Comment by dsherret** [comment]
@kt3k this is in a much better state to add redirects now I think. I did a few refactors in a separate PR and some in here. Let me know if you have any questions (there are a couple failing tests right now for parsing)

**Review by bartlomieju — APPROVED** [review]
LGTM, please remember to update manual before landing

File: src/shell.rs [hunk 1]
```diff
@@ -248,23 +251,115 @@ async fn execute_pipeline_inner(
...
+      ),
+      words.len(),
+      words.join(" ")
+    ));
+    return Err(ExecuteResult::from_exit_code(1));
+  }
+  let output_path = &words[0];
+
+  // cross platform supress output
+  if output_path == "/dev/null" {
```
Comment (kt3k) [inline src/shell.rs:321]: Nice to have cross platform `/dev/null` 👍

**Review by kt3k — APPROVED** [review]
LGTM with nits.

Nice!

**Pull request merged**